test/framework/src/main/java/org/elasticsearch/test/InternalTestCluster.java

```diff
@@ -1667,141 +1714,23 @@ private int getMasterNodesCount() {
         return startNode(settings1);
     }
 
-    /**
-     * Starts a node in an async manner with the given settings and returns future with its name.
-     */
-    public synchronized Async<String> startNodeAsync() {
-        return startNodeAsync(Settings.EMPTY);
-    }
-
-    /**
-     * Starts a node in an async manner with the given settings and returns future with its name.
-     */
-    public synchronized Async<String> startNodeAsync(final Settings settings) {
-        final int defaultMinMasterNodes;
-        if (autoManageMinMasterNodes) {
-            int mastersDelta = Node.NODE_MASTER_SETTING.get(settings) ? 1 : 0;
-            defaultMinMasterNodes = updateMinMasterNodes(getMasterNodesCount() + mastersDelta);
-        } else {
-            defaultMinMasterNodes = -1;
-        }
-        return startNodeAsync(settings, defaultMinMasterNodes);
-    }
-
-    private synchronized Async<String> startNodeAsync(final Settings settings, int defaultMinMasterNodes) {
-        final NodeAndClient buildNode = buildNode(settings, defaultMinMasterNodes);
-        final Future<String> submit = executor.submit(() -> {
-            buildNode.startNode();
-            publishNode(buildNode);
-            return buildNode.name;
-        });
-        return () -> submit.get();
-    }
-
-
-    /**
-     * Starts multiple nodes in an async manner and returns future with its name.
-     */
-    public synchronized Async<List<String>> startNodesAsync(final int numNodes) {
-        return startNodesAsync(numNodes, Settings.EMPTY);
-    }
-
-    /**
-     * Starts multiple nodes in an async manner with the given settings and returns future with its name.
-     */
-    public synchronized Async<List<String>> startNodesAsync(final int numNodes, final Settings settings)  {
-        final int defaultMinMasterNodes;
-        if (autoManageMinMasterNodes) {
-           int mastersDelta = Node.NODE_MASTER_SETTING.get(settings) ? numNodes : 0;
-            defaultMinMasterNodes = updateMinMasterNodes(getMasterNodesCount() + mastersDelta);
-        } else {
-            defaultMinMasterNodes = -1;
-        }
-        final List<Async<String>> asyncs = new ArrayList<>();
-        for (int i = 0; i < numNodes; i++) {
-            asyncs.add(startNodeAsync(settings, defaultMinMasterNodes));
-        }
-
-        return () -> {
-            List<String> ids = new ArrayList<>();
-            for (Async<String> async : asyncs) {
-                ids.add(async.get());
-            }
-            return ids;
-        };
-    }
-
-    /**
-     * Starts multiple nodes (based on the number of settings provided) in an async manner, with explicit settings for each node.
-     * The order of the node names returned matches the order of the settings provided.
-     */
-    public synchronized Async<List<String>> startNodesAsync(final Settings... settings) {
```

Comment:
Can the `Async` interface also be removed? I see it's an interface defined inside InternTestCluster.